# Change peak table form

In this notebook, we will change the form of the peak table to have the variables in columns and samples in rows. We will also add the groups (Liver Cancer vs. Case control) as a column with the metadata text file.

---

# Import modules

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

In [2]:
import re
pwd = ! pwd
pwd = re.sub('notebooks/notebooks_prepare_HCC_data', '', pwd[0])
pwd

'/home/jovyan/work/'

# Import peak table HILIC POS

In [3]:
peakTable_HILIC_POS = pd.read_csv(pwd + 'data/HILIC_POS_results/process_output_quantid_pos_camera_metfrag/peaktablePOSout_pos_metfrag.txt', sep='\t', decimal='.', na_values='NA')
peakTable_HILIC_POS.head()

dataMatrix  EPIC_Liver_Cancer_NR160809_007_41_LivCan_153_007.mzML  \
0  variable_3                                          19.761786       
1  variable_5                                          14.635833       
2  variable_6                                          22.185582       
3  variable_7                                          16.414744       
4  variable_8                                          12.615143       

   EPIC_Liver_Cancer_NR160809_008_41_LivCan_154_008.mzML  \
0                                          19.735279       
1                                          15.193396       
2                                          20.831414       
3                                          15.052533       
4                                                NaN       

   EPIC_Liver_Cancer_NR160809_009_41_LivCan_267_009.mzML  \
0                                          19.116076       
1                                          15.446714       
2                                          22.152604       
3                                          16.410827       
4                                                NaN       

   EPIC_Liver_Cancer_NR160809_010_41_LivCan_268_010.mzML  \
0                                          19.376875       
1                                          15.460525       
2                                          22.169389       
3                                          16.403964       
4                                          12.720767       

   EPIC_Liver_Cancer_NR160809_014_21_LivCan_299_014.mzML  \
0                                          19.555938       
1                                          14.515536       
2                                          22.150249       
3                                          16.372219       
4                                          12.314019       

   EPIC_Liver_Cancer_NR160809_015_21_LivCan_300_015.mzML  \
0                                          19.377437       
1                                          14.913215       
2                                          22.198844       
3                                          16.428678       
4                                                NaN       

   EPIC_Liver_Cancer_NR160809_018_22_LivCan_085_018.mzML  \
0                                          19.526568       
1                                          14.648653       
2                                          22.092599       
3                                          16.311085       
4                                          12.472396       

   EPIC_Liver_Cancer_NR160809_019_22_LivCan_086_019.mzML  \
0                                          19.349326       
1                                          14.454301       
2                                          22.093306       
3                                          16.302336       
4                                                NaN       

   EPIC_Liver_Cancer_NR160809_020_25_LivCan_309_020.mzML  ...  \
0                                          19.610237      ...   
1                                          14.713504      ...   
2                                          22.006737      ...   
3                                          16.196649      ...   
4                                          12.438756      ...   

   EPIC_Liver_Cancer_NR160809_270_35_LivCan_161_270.mzML  \
0                                          19.289850       
1                                          13.975330       
2                                          21.387145       
3                                          15.588615       
4                                          12.545073       

   EPIC_Liver_Cancer_NR160809_271_35_LivCan_162_271.mzML  \
0                                          18.755673       
1                                          14.763973       
2                                          20.905625       
3                                          15.15768

In [4]:
peakTable_HILIC_POS.shape

(557, 187)

The peak table contains variables in rows (557 variables) and samples in columns (187 samples).

# Import sample metadata

In [5]:
metadata_HILIC_POS = pd.read_csv(pwd + 'data/HILIC_POS_results/process_output_quantid_pos_camera_metfrag/metadataPOSout_pos_metfrag.txt', sep='\t', decimal='.', na_values='NA')
metadata_HILIC_POS.head()

sampleMetadata    Class    Groups   X
0  EPIC_Liver_Cancer_NR160809_007_41_LivCan_153_0...  Samples  Incident NaN
1  EPIC_Liver_Cancer_NR160809_008_41_LivCan_154_0...  Samples  Non_case NaN
2  EPIC_Liver_Cancer_NR160809_009_41_LivCan_267_0...  Samples  Incident NaN
3  EPIC_Liver_Cancer_NR160809_010_41_LivCan_268_0...  Samples  Non_case NaN
4  EPIC_Liver_Cancer_NR160809_014_21_LivCan_299_0...  Samples  Incident NaN

In [6]:
metadata_HILIC_POS.shape

(186, 4)

In [7]:
metadata_HILIC_POS.insert(1, 'SampleID', ['_'.join(filename.split('_')[6:8]) for filename in metadata_HILIC_POS['sampleMetadata']])

In [8]:
metadata_HILIC_POS

sampleMetadata    SampleID    Class  \
0    EPIC_Liver_Cancer_NR160809_007_41_LivCan_153_0...  LivCan_153  Samples   
1    EPIC_Liver_Cancer_NR160809_008_41_LivCan_154_0...  LivCan_154  Samples   
2    EPIC_Liver_Cancer_NR160809_009_41_LivCan_267_0...  LivCan_267  Samples   
3    EPIC_Liver_Cancer_NR160809_010_41_LivCan_268_0...  LivCan_268  Samples   
4    EPIC_Liver_Cancer_NR160809_014_21_LivCan_299_0...  LivCan_299  Samples   
..                                                 ...         ...      ...   
181  EPIC_Liver_Cancer_NR160809_275_32_LivCan_080_2...  LivCan_080  Samples   
182  EPIC_Liver_Cancer_NR160809_281_22_LivCan_375_2...  LivCan_375  Samples   
183  EPIC_Liver_Cancer_NR160809_282_22_LivCan_376_2...  LivCan_376  Samples   
184  EPIC_Liver_Cancer_NR160809_283_71_LivCan_165_2...  LivCan_165  Samples   
185  EPIC_Liver_Cancer_NR160809_284_71_LivCan_166_2...  LivCan_166  Samples   

       Groups   X  
0    Incident NaN  
1    Non_case NaN  
2    Incident NaN  
3    Non_case NaN  
4    Incident NaN  
..        ...  ..  
181  Non_case NaN  
182  Incident NaN  
183  Non_case NaN  
184  Incident NaN  
185  Non_case NaN  

[186 rows x 5 columns]

In [9]:
EPIC_Liver_sample_informations = pd.read_csv(pwd + 'data/metadata/EPIC-Liver_sample_informations.csv')
EPIC_Liver_sample_informations = EPIC_Liver_sample_informations[EPIC_Liver_sample_informations['Id Metabo'].isin(metadata_HILIC_POS['SampleID'])]
EPIC_Liver_sample_informations = EPIC_Liver_sample_informations.set_index('Id Metabo')
EPIC_Liver_sample_informations = EPIC_Liver_sample_informations.reindex(index=metadata_HILIC_POS['SampleID'])
EPIC_Liver_sample_informations = EPIC_Liver_sample_informations.reset_index()
EPIC_Liver_sample_informations = EPIC_Liver_sample_informations[['SampleID', 'CaseCtrl', 'Match_Caseset', 'Typ_Tumo', 'Case_HCC_Wide', 'Idepic', 'Idepic_Bio', 'Id Aliquot']]
EPIC_Liver_sample_informations = EPIC_Liver_sample_informations.rename(columns={'CaseCtrl': 'Groups', 'Match_Caseset': 'MatchCaseset', 'Typ_Tumo': 'TypTumo', 'Case_HCC_Wide': 'CaseHCCWide', 'Idepic': 'EPICID', 'Idepic_Bio': 'EPICBioID', 'Id Aliquot': 'AliquotID'})
EPIC_Liver_sample_informations['TypTumo'] = EPIC_Liver_sample_informations['TypTumo'].str.replace('/', '')
EPIC_Liver_sample_informations

SampleID    Groups  MatchCaseset   TypTumo CaseHCCWide          EPICID  \
0    LivCan_153  Incident            82       HCC         Yes  41____41041872   
1    LivCan_154  Non-case            82       NaN          No  41____41046868   
2    LivCan_267  Incident           427       HCC         Yes  41____41074588   
3    LivCan_268  Non-case           427       NaN          No  41____41089214   
4    LivCan_299  Incident           376  HCC_Wide         Yes  21____21103888   
..          ...       ...           ...       ...         ...             ...   
181  LivCan_080  Non-case            56       NaN          No  32____3208004E   
182  LivCan_375  Incident           383       HCC         Yes  22____22281416   
183  LivCan_376  Non-case           383       NaN          No  22____22281701   
184  LivCan_165  Incident           147       HCC         Yes  71____71000825   
185  LivCan_166  Non-case           147       NaN          No  71____71001996   

          EPICBioID      AliquotID  
0    41____41041872  41041872-s2#C  
1    41____41046868  41046868-s2#C  
2    41____41074588  41074588-s1#B  
3    41____41089214  41089214-s1#B  
4    21____21103888  21103888-s1#B  
..              ...            ...  
181  32____32253231  32253231-s3#C  
182  22____22281416  22281416-s1#B  
183  22____22281701  22281701-s1#B  
184  71____71000065  71000065-s2#C  
185  71____71000216  71000216-s2#C  

[186 rows x 8 columns]

# Change form of the peak table

In [10]:
# Keep variable names in a separate list for the column names
variable_names = peakTable_HILIC_POS['dataMatrix'].values

# Transpose the peak table
peakTable_HILIC_POS = peakTable_HILIC_POS.iloc[:,1:].transpose()

# Reset the indexes
peakTable_HILIC_POS = peakTable_HILIC_POS.reset_index()

# Add a column with simplified file names
peakTable_HILIC_POS.insert(1, 'SampleID', ['_'.join(filename.split('_')[6:8]) for filename in metadata_HILIC_POS['sampleMetadata']])

# Delete column with full file name
peakTable_HILIC_POS = peakTable_HILIC_POS.iloc[:,2:]

# Add column names with variables
peakTable_HILIC_POS.columns = variable_names

In [11]:
peakTable_HILIC_POS

variable_3  variable_5  variable_6  variable_7  variable_8  variable_10  \
0     19.761786   14.635833   22.185582   16.414744   12.615143    15.580604   
1     19.735279   15.193396   20.831414   15.052533         NaN    15.145423   
2     19.116076   15.446714   22.152604   16.410827         NaN    14.628890   
3     19.376875   15.460525   22.169389   16.403964   12.720767    15.937169   
4     19.555938   14.515536   22.150249   16.372219   12.314019    14.950203   
..          ...         ...         ...         ...         ...          ...   
181   19.664978   14.447858   20.808503   15.027154   12.648327    14.806952   
182   19.301954   14.839941   21.085175   15.321075   12.510670    14.488147   
183   19.170626   14.291711   21.381912   15.197205   12.499558    14.434959   
184   18.869505   14.440576   20.961485   14.829978   12.312338    14.114074   
185   18.954904         NaN   20.005032         NaN   12.414258          NaN   

     variable_11  variable_14  variable_22  variable_23  ...  variable_2339  \
0      15.881437    17.500899    12.792893    16.805604  ...      12.052252   
1      14.034654    17.953780          NaN    15.907714  ...      13.020365   
2      15.316772    17.404715    12.786407    15.947176  ...            NaN   
3      15.967749    15.836957          NaN    16.371211  ...            NaN   
4      15.294906    16.610171    12.214057    16.171763  ...            NaN   
..           ...          ...          ...          ...  ...            ...   
181    14.244917    17.126579    12.080710    15.388376  ...      12.349299   
182    15.163197    16.816591    12.452644    16.474456  ...      12.909283   
183    14.520558    16.315887    12.808417    15.613263  ...      12.846355   
184    13.776358    16.110117    12.667428    16.021372  ...      12.477753   
185          NaN    15.432975    12.047361    15.443640  ...      13.225899   

     variable_2340  variable_2350  variable_2353  variable_2364  \
0        12.568647      13.192848      13.230752      12.248590   
1        12.928643      13.697314            NaN      12.604052   
2              NaN            NaN      13.330175            NaN   
3              NaN      13.086805      13.304171            NaN   
4              NaN            NaN      13.306722      12.349830   
..             ...            ...            ...            ...   
181      11.505004      13.189121            NaN            NaN   
182      12.642143            NaN            NaN      12.708850   
183      13.243400            NaN            NaN      12.199647   
184      11.789382            NaN            NaN      11.628154   
185      12.978890      11.596877            NaN            NaN   

     variable_2365  variable_2368  variable_2370  variable_2375  variable_2379  
0        12.114328      14.269258      12.355017      13.114289      13.882526  
1              NaN            NaN      12.208084            NaN            NaN  
2              NaN      14.215988            NaN      12.955618      13.778398  
3        11.605353      14.284148      12.319119      13.058285      13.812829  
4        11.926237      14.192775            NaN      13.006556      13.841201  
..             ...            ...            ...            ...            ...  
181            NaN            NaN            NaN            NaN            NaN  
182      12.462751      12.175110      12.635066            NaN            NaN  
183            NaN            NaN      12.258149            NaN            NaN  
184            NaN            NaN            NaN            NaN            NaN  
185            NaN            NaN      12.158923            NaN            NaN  

[186 rows x 557 columns]

# Prepare data to save it as csv

In [12]:
peakTable_HILIC_POS_full = EPIC_Liver_sample_informations.join(peakTable_HILIC_POS)
print(peakTable_HILIC_POS_full.shape)
peakTable_HILIC_POS_full.head()

(186, 565)


SampleID    Groups  MatchCaseset   TypTumo CaseHCCWide          EPICID  \
0  LivCan_153  Incident            82       HCC         Yes  41____41041872   
1  LivCan_154  Non-case            82       NaN          No  41____41046868   
2  LivCan_267  Incident           427       HCC         Yes  41____41074588   
3  LivCan_268  Non-case           427       NaN          No  41____41089214   
4  LivCan_299  Incident           376  HCC_Wide         Yes  21____21103888   

        EPICBioID      AliquotID  variable_3  variable_5  ...  variable_2339  \
0  41____41041872  41041872-s2#C   19.761786   14.635833  ...      12.052252   
1  41____41046868  41046868-s2#C   19.735279   15.193396  ...      13.020365   
2  41____41074588  41074588-s1#B   19.116076   15.446714  ...            NaN   
3  41____41089214  41089214-s1#B   19.376875   15.460525  ...            NaN   
4  21____21103888  21103888-s1#B   19.555938   14.515536  ...            NaN   

   variable_2340  variable_2350  variable_2353  variable_2364  variable_2365  \
0      12.568647      13.192848      13.230752      12.248590      12.114328   
1      12.928643      13.697314            NaN      12.604052            NaN   
2            NaN            NaN      13.330175            NaN            NaN   
3            NaN      13.086805      13.304171            NaN      11.605353   
4            NaN            NaN      13.306722      12.349830      11.926237   

   variable_2368  variable_2370  variable_2375  variable_2379  
0      14.269258      12.355017      13.114289      13.882526  
1            NaN      12.208084            NaN            NaN  
2      14.215988            NaN      12.955618      13.778398  
3      14.284148      12.319119      13.058285      13.812829  
4      14.192775            NaN      13.006556      13.841201  

[5 rows x 565 columns]

# Save cleaned peak table in csv file

In [14]:
peakTable_HILIC_POS_full.to_csv(pwd + 'data/prepare_HCC_data/peakTable_HILIC_POS_full.csv', index = False, header=True)